sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriver.java

```diff
@@ -141,17 +144,34 @@ public Connection connect(String url, Properties info) throws SQLException {
    * not this path. The CLI ends up using the schema factory that populates the default schema with
    * all table providers it can find. See {@link BeamCalciteSchemaFactory}.
    */
-  public static JdbcConnection connect(TableProvider tableProvider) {
+  public static JdbcConnection connect(TableProvider tableProvider, PipelineOptions options) {
     try {
       Properties properties = new Properties();
       properties.setProperty(
           SCHEMA_FACTORY.camelName(), BeamCalciteSchemaFactory.Empty.class.getName());
       JdbcConnection connection =
           (JdbcConnection) INSTANCE.connect(CONNECT_STRING_PREFIX, properties);
       connection.setSchema(TOP_LEVEL_BEAM_SCHEMA, tableProvider);
+      connection.setPipelineOptionsMap(getOptionsMap(options));
       return connection;
     } catch (SQLException e) {
       throw new RuntimeException(e);
     }
   }
+
+  /** Converts {@link PipelineOptions} to its map format. */
+  private static Map<String, String> getOptionsMap(PipelineOptions options) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    Map map = objectMapper.convertValue(options, Map.class);
+    map = (Map) map.get("options");
+    if (map == null) {
+      map = new HashMap();
+    }
+    Map<String, String> map2 = new HashMap<>();
+    for (Object entry : map.entrySet()) {
+      Map.Entry ent = (Map.Entry) entry;
+      map2.put(ent.getKey().toString(), ent.getValue().toString());
```

Comment:
`getValue().toString()` is probably wrong. I think if you have a complex object there, it would be converted to a `Map<>` by the `ObjectMapper` and then `toString()` would return something like `{key=value}` which is not a valid JSON, so we won't be able to parse it back. What has to be done is we should use the same mapper to convert the value to JSON string instead, probably something along these lines:

```java
optionsMap = map.entrySet()
 .stream()
 .collect(toMap(MapEntry::getKey, e-> OBJECT_MAPPER.writeValueAsString(e.getValue())))
```